### PR TITLE
Add contribution page tokens to contribution tokens

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Add contribution page tokens to contribution tokens

Before
----------------------------------------
Tokens not available 

After
----------------------------------------
A small handful are

Technical Details
----------------------------------------
@larssandergreen this is a reviewers cut of the token portion of the changes in https://github.com/civicrm/civicrm-core/pull/27173 - the main change is that I added some extra tokens that I could see would be required in the online contribution template

My other focus was to work through the code & understand where the html handling would best be done. I have some thoughts based on this - which I will put up separately. Note the test I put up is just where I was most comfortable putting it - I suggest we add both this one  & your test tweaks (once this all passes)

Comments
----------------------------------------
Update - I think the line break handling best lives in the coversion within the token - ie https://github.com/civicrm/civicrm-core/pull/27204 and #27203

### CODE REVIEW

File: CRM/Core/EntityTokens.php [hunk 1]
```diff
@@ -138,9 +138,23 @@ public function evaluateToken(TokenRow $row, $entity, $field, $prefetch = NULL)
         Civi::log()->info('invalid date token');
       }
     }
+    if ($this->isHTMLTextField($field)) {
```
Comment (eileenmcnaughton) [inline CRM/Core/EntityTokens.php:141]: So we are interpretting the field metadata to tell the token system if we are giving it html or plain text

File: tests/phpunit/CRM/Utils/TokenConsistencyTest.php [hunk 2]
```diff
@@ -195,11 +195,48 @@ public function testContributionRecurTokenConsistency(): void {
...
+    $this->contributionPageCreate($tokenValues);
+
+    $tokenProcessor = $this->getTokenProcessor(['schema' => ['contributionId']]);
+    $tokenProcessor->addMessage('text', $tokenString, 'text/plain');
+    $tokenProcessor->addMessage('html', $tokenString, 'text/html');
+    $tokenProcessor->addRow(['contributionId' => $this->contributionCreate(['contribution_page_id' => $this->ids['ContributionPage']['test'], 'contact_id' => $this->individualCreate()])]);
+    $tokenProcessor->evaluate();
+    $this->assertEquals('contribution.contribution_page_id.frontend_title :public title
+contribution.contribution_page_id.pay_later_text :pay later text
+contribution.contribution_page_id.pay_later_receipt :&lt;p&gt;first line&lt;/p&gt;&lt;p&gt;second line&lt;/p&gt;
```
Comment (eileenmcnaughton) [inline tests/phpunit/CRM/Utils/TokenConsistencyTest.php:227]: Here we can see the token system html-ises plain when rendering html & renders the same text raw for plain. Which is as you suggested was the case

File: tests/phpunit/CRM/Utils/TokenConsistencyTest.php [hunk 3]
```diff
@@ -195,11 +195,48 @@ public function testContributionRecurTokenConsistency(): void {
...
+contribution.contribution_page_id.is_share :1
+contribution.contribution_page_id.receipt_text :Text in
+ non html', $tokenProcessor->getRow(0)->render('html'));
```
Comment (eileenmcnaughton) [inline tests/phpunit/CRM/Utils/TokenConsistencyTest.php:230]: html version NOT parsed through nlt2br

**Comment by colemanw** [comment]
Test cover looks good

**Pull request merged**